packages/jest-config/src/normalize.js

```diff
@@ -52,19 +52,27 @@ const setupPreset = (
 ): InitialOptions => {
   let preset;
   const presetPath = _replaceRootDirInPath(options.rootDir, optionsPreset);
-  const presetModule = Resolver.findNodeModule(
-    presetPath.endsWith(JSON_EXTENSION)
-      ? presetPath
-      : path.join(presetPath, PRESET_NAME),
-    {
-      basedir: options.rootDir,
-    },
-  );
+  const foundModule = PRESET_EXTENSIONS.some(ext => {
+    const presetModule = Resolver.findNodeModule(
+      presetPath.charAt(0) === '.' || presetPath.endsWith(ext)
+        ? presetPath
+        : path.join(presetPath, PRESET_NAME + ext),
+      {
+        basedir: options.rootDir,
+      },
+    );
+
+    try {
+      // $FlowFixMe
+      preset = (require(presetModule): InitialOptions);
```

Comment:
I worry about this not working with multiple projects, you'd want the preset to be re-evaluated every time. Can we remove it from the require.cache right after requiring it? :)

Reply:
And guard that in a separate try-catch, I don't want Jest to rely on require.cache to be present :D